### PR TITLE
fix 35165, salt-run jobs.exit_success jid is broken

### PR DESCRIPTION
**Pull request by hu-dabao**
### What does this PR do?

Fix https://github.com/saltstack/salt/issues/35165
For more details about the issue, please check the issue url.

A little bit more details about the fix:
It is calling `list_job` rather than `lookup_jid` because `lookup_jid` only returns successful `'Result'` of returned minions whereas `list_job` returns more info which include failed minions as well. 
From the load perspective, calling `list_job` is same as calling `lookup_jid` because `lookup_jid` just calls `list_job` and trim the `'Result'`. 

`exit_success` will return `True` for minions that have successful returns and `False` for those that don't have returns or defunct returns. 

`retcode` is not able to be used in determining job success because all of the job management and execution codes do not have `retcode` as a field. Feel free to correct me if I am wrong
### Previous Behavior

Check the issue for more details. 
In short, `salt-run jobs.exit_success jid` is broken
### New Behavior

```
salt-run jobs.exit_success jid
minion1:
    False
minion2:
    True
```
### Tests written?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

**Comment by babilen** [comment]
Thank you!

**Comment by evan-ty** [comment]
Hey guys, thanks for looking into this.  I was using this via the salt runner client api, and I noticed the behavior that it reports that the run was a success even though some states exited with a `retcode` of 1:

https://paste.debian.net/810250/

In this paste, there is the output and the code I used to get that output.
